### PR TITLE
S1 t5/integrate frontend and backend for signup functionality

### PR DESCRIPTION
**Pull request by sehersarin**
**Scrum Board Ticket:** S1T5: Integrate frontend and backend for the signup functionality
**Story:** Story 1 - Sign up using email 

**Description:** Updated the logic and functionality for the signup form and integrated it with the backend. 

- Adjusted the navbar in the login and signup pages to be of the same height as the one in the system as per TA comments in Sprint 2 demo.
- Marked required fields with * to show priorities and added in error handling messages to let user know that input is mandatory as per TA comments in Sprint 2 demo.
- Integrated frontend with backend by calling the /create-user/ backend api on click and passing in the required parameters.
- Frontend unit tests were also written to validate this ticket.

The signup form enables users to enter their first and last name, select user role, worker type and specialization, enter ID, email, phone number and password to create an account in the system. 

**Testing Criteria (how I tested the code):** 
1. Tested UI of Signup form by running locally and verifying the front-end has matched design specifications (adjusted nav bar, indication of required fields.) 
2. Tested form input criteria for a variety of instances including:
    1. Not inputting a mandatory field (ex. First name, last name, email etc.) and trying to submit form.
    2. Entering all form criteria and submitting the form.
3. Ran the automated tests in the react-ui folder and ensured that there was a high file coverage and all tests passed.

**How Reviewers can Test Code:**
1. Run the code locally and ensure no complications when starting up.
2. Navigate inside the `react-ui` and `server` folders and run npm start.
3. View the screens and UI to ensure it looks like design specifications (adjusted nav bar, indication of required fields.)

Screen Shot 2020-11-18 at 11 31 07 AM

4. Attempt to leave a mandatory field blank (ie. email) when trying to submit form and view message to “Please fill out this field.”
5. Inside the `react-ui` folder, run` npm test`. All tests should pass and the SignupForm component should have 90% file coverage - as seen in the screenshot below (the rest are not set up yet and are in the scope for another ticket.)

Screen Shot 2020-11-18 at 11 27 47 AM

Please note - upon correct form submission this screen currently appears:

Screen Shot 2020-11-18 at 11 31 39 AM

### CODE REVIEW

File: react-ui/src/components/SignUpForm.js [hunk 1]
```diff
@@ -37,52 +45,120 @@ handleFormChange = event => {
   console.log(stateName, val);
 };
 
-handleSubmit(e) {
-    e.preventDefault();
+signup = event => {
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:48]: passed in required parameters for integration and updated variable names to lowerCamel case for consistency.

Comment (prdbondi) [reply react-ui/src/components/SignUpForm.js:48]: Thank you for the attention to detail, Seher!

File: react-ui/src/components/Header.scss [hunk 2]
```diff
@@ -6,8 +6,8 @@
   
   .App-header-blue {
     background-color: #eae8ff;
-    height: 40px;
-    /* padding: 5px; */
+    height: 70px;
```
Comment (prdbondi) [inline react-ui/src/components/Header.scss:9]: I see this fix is in response to TA feedback from the demo, well done!

File: server/routes/public.js [hunk 3]
```diff
@@ -43,7 +43,7 @@ router.get('/create-user', async (req, res) => {
         schoolId: Joi.number().positive().integer(),
         userType: Joi.string().min(1).valid('student', 'worker'),
         type: Joi.string(),
-        specialization: Joi.string().valid('social worker', 'guidance councellor'),//verify options
+        specialization: Joi.string(),//verify options
```
Comment (sehersarin) [inline server/routes/public.js:46]: Updated backend code to ensure correct value was called for variable.

File: react-ui/src/components/Header.scss (hunk 2 above)
Comment (sehersarin) [reply react-ui/src/components/Header.scss:9]: Yes it is, thank you Peter!

File: server/routes/public.js (hunk 3 above)
Comment (prdbondi) [reply server/routes/public.js:46]: Understood!

File: react-ui/src/components/SignUpForm.js [hunk 4]
```diff
@@ -37,52 +45,120 @@ handleFormChange = event => {
...
                 <label>
                 <input 
                   className ="InputFields" 
                   type="text" 
-                  name="first_name"
+                  name="firstName"
                   size ="40"
-                  placeholder= "Enter First Name" 
+                  placeholder= "Enter First Name*" 
+                  required="required"
```
Comment (sehersarin) [inline react-ui/src/components/SignUpForm.js:136]: Ensured specific fields have a "required" attribute.

**Review by prdbondi — APPROVED** [review]
image

Seher, I have followed testing steps 1-5 and all steps are successful! 
See screenshot related to successful testing of step 4 f the testing process.

**Comment by sehersarin** [comment]
Great, glad that all steps worked! Thank you for the review, Peter!

**Pull request merged**